### DISCUSSION TITLE
bad gateway 502 for localhost

### DISCUSSION THREAD

**Discussion by gkucmierz** [NginxProxyManager/nginx-proxy-manager#3279] [Q&A]
I am trying to setup proxy hosts for various applications running on docker.

But whenever I use `localhost` or `127.0.01` host Then I have bad gateway error.

When I set host as machine ip address in my local network `192.168.x.x` then it works, but it requires from me allow opened ports on `ufw` and also I am assuming it produces unnecessary traffic in my local network.

What can cause this problem?

**Comment by maurxce** [comment]
Have you tried connecting the containers via a network and then using their respective hostnames?

**Reply by gkucmierz** [reply]
No, I am not sure how to do that yet.

**Reply by maurxce** [reply]
I haven't used it myself yet, but it should be possible.
I'd recommend reading this: [Networking in Compose](https://docs.docker.com/compose/networking/) assuming you're using compose files

**Comment by blazejwiecha** [comment]
So are here any way to resolve error? 
In my case all works till go to https ... then error gateway 
on http works ok

**Reply by kpleines** [reply]
Same Problem here. 
I've installed a fresh nginx prody manager an login works fine on first startup.
After a reboot I got the "Bad Gateway" on login. Proxy feature itself works fine.

I use IP for login to npm.
